packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -38,6 +38,8 @@ export enum TelemetryDataTag {
     UserData = "UserData",
 }
 
+export type TelemetryEventPropertyTypeExt = string | number | boolean | undefined | (string | number | boolean)[];
```

Comment:
Might want to support null as well here, given you are making these changes. We actually pass null all the time (ISequencedDocumentMessage.clientId can be null, despite type saying otherwise - just opened ticket to fix it).

You probably want to allow undefined to be element of array either.
Though this is slippery, as it will show up as null after stringify / parse cycle.